src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -187,10 +188,15 @@
   "Returns SQL for performing a regexp match against the contents of
   an array. If any of the array's items match the supplied regexp,
   then that satisfies the match."
-  [orig-table _ column]
+  [column]
   (hcore/raw
    (format "EXISTS(SELECT 1 FROM UNNEST(%s) WHERE UNNEST ~ ?)" (name column))))
 
+(defn sql-in-array
+  [column]
+  (hcore/raw
+   (format "EXISTS(SELECT 1 from UNNEST(?) WHERE unnest = %s)" (name column))))
```

Comment:
I thought we didn't want to use that because ANY didn't use indices:

```bash
puppetdb=# explain select * from reports where exists(select * from unnest(ARRAY[1,2]) inner join reports on unnest.unnest = reports.id);
                                               QUERY PLAN
---------------------------------------------------------------------------------------------------------
 Result  (cost=1.46..11.96 rows=50 width=1347)
   One-Time Filter: $1
   InitPlan 1 (returns $1)
     ->  Nested Loop  (cost=0.14..66.00 rows=50 width=0)
           ->  Function Scan on unnest  (cost=0.00..1.00 rows=100 width=4)
           ->  Index Only Scan using reports_pkey on reports reports_1  (cost=0.14..0.64 rows=1 width=8)
                 Index Cond: (id = unnest.unnest)
   ->  Seq Scan on reports  (cost=0.00..10.50 rows=50 width=1347)
(8 rows)

Time: 1.117 ms
puppetdb=# explain select * from reports where reports.id = ANY(ARRAY[1,2]);
                        QUERY PLAN
-----------------------------------------------------------
 Seq Scan on reports  (cost=0.00..10.62 rows=2 width=1347)
   Filter: (id = ANY ('{1,2}'::integer[]))
(2 rows)
```